chia/wallet/wallet_nft_store.py

```diff
@@ -0,0 +1,140 @@
+from chia.util.db_wrapper import DBWrapper
+from chia.util.ints import uint32
+from chia.wallet.lineage_proof import LineageProof
+from chia.wallet.nft_wallet.nft_info import DEFAULT_STATUS, IN_TRANSACTION_STATUS, NFTCoinInfo
+
+_T_WalletNftStore = TypeVar("_T_WalletNftStore", bound="WalletNftStore")
+
+
+class WalletNftStore:
+    """
+    WalletNftStore keeps track of all user created NFTs and necessary smart-contract data
+    """
+
+    db_connection: aiosqlite.Connection
+    db_wrapper: DBWrapper
+
+    @classmethod
+    async def create(cls: Type[_T_WalletNftStore], db_wrapper: DBWrapper) -> _T_WalletNftStore:
+        self = cls()
+        self.db_wrapper = db_wrapper
+        self.db_connection = db_wrapper.db
+        await self.db_connection.execute(
+            (
+                "CREATE TABLE IF NOT EXISTS users_nfts("
+                " nft_id text PRIMARY KEY,"
+                " nft_coin_id text,"
+                " wallet_id int,"
+                " did_id text,"
+                " coin text,"
+                " lineage_proof text,"
+                " mint_height bigint,"
+                " status text,"
+                " full_puzzle blob)"
+            )
+        )
+        await self.db_connection.execute("CREATE INDEX IF NOT EXISTS nft_coin_id on users_nfts(nft_coin_id)")
+        await self.db_connection.execute("CREATE INDEX IF NOT EXISTS nft_wallet_id on users_nfts(wallet_id)")
+        await self.db_connection.execute("CREATE INDEX IF NOT EXISTS nft_did_id on users_nfts(did_id)")
+        await self.db_connection.commit()
+        return self
+
+    async def _clear_database(self) -> None:
+        cursor = await self.db_connection.execute("DELETE FROM users_nfts")
+        await cursor.close()
+        await self.db_connection.commit()
+
+    async def delete_nft(self, nft_id: bytes32, in_transaction: bool = False) -> None:
+        if not in_transaction:
+            await self.db_wrapper.lock.acquire()
+        try:
+            cursor = await self.db_connection.execute(f"DELETE FROM users_nfts where nft_id='{nft_id.hex()}'")
+            await cursor.close()
+        finally:
+            if not in_transaction:
+                await self.db_connection.commit()
+                self.db_wrapper.lock.release()
+
+    async def save_nft(
+        self, wallet_id: uint32, did_id: Optional[bytes32], nft_coin_info: NFTCoinInfo, in_transaction: bool = False
+    ) -> None:
+        if not in_transaction:
+            await self.db_wrapper.lock.acquire()
+        try:
+            cursor = await self.db_connection.execute(
+                "INSERT or REPLACE INTO users_nfts VALUES(?, ?, ?, ?, ?, ?, ?, ?, ?)",
+                (
+                    nft_coin_info.nft_id.hex(),
+                    nft_coin_info.coin.name().hex(),
+                    int(wallet_id),
+                    did_id.hex() if did_id else None,
+                    json.dumps(nft_coin_info.coin.to_json_dict()),
```

Comment:
why not store all fields you're transforming into json or hex string into bytes? It saves space and avoids serialisation. Like you did for the`full_puzzle` field?

Reply:
For lineage_proof and full_puzzle it make sense, but for other fields, they should be queryable. I think it's not worth to create another class just have lineage_proof and full_puzzle.

Reply:
ok, fair point